### PR TITLE
Require Microsoft.VisualStudio.Component.VC.Tools.x86.x64 to be installed before compiling ANCM

### PR DESCRIPTION
**Pull request by jkotalik**
Reacting to https://github.com/aspnet/BuildTools/commit/ec1822a2c89593143f9b8deb7a56215724177955

### CODE REVIEW

File: build/repo.targets [hunk 1]
```diff
@@ -4,26 +4,18 @@
...
-
-  <Target Name="BuildNativeAssets" DependsOnTargets="_LocateMSBuildExe" BeforeTargets="Compile">
+  <Target Name="BuildNativeAssets" DependsOnTargets="GetToolsets" BeforeTargets="Compile" >
     <ItemGroup>
       <BuildConfigurations Include="/p:Configuration=Release /p:platform=Win32" />
       <BuildConfigurations Include="/p:Configuration=Release /p:platform=x64" />
     </ItemGroup>
-    <Exec Command="&quot;$(MSBuildExePath)&quot; &quot;$(RepositoryRoot)src\AspNetCore\AspNetCore.vcxproj&quot; %(BuildConfigurations.Identity)" />
+
+    <Error
```
Comment (jkotalik) [inline build/repo.targets:13]: @natemcmaster Error or warning here?

Comment (natemcmaster) [reply build/repo.targets:13]: IMO error. You can't build anything useful in this project without it.

File: build/repo.targets [hunk 2]
```diff
@@ -4,26 +4,18 @@
...
+      Text="Unable to locate MSBuild 15.0 under $(VisualStudioMSBuildx86Path)"
```
Comment (natemcmaster) [inline build/repo.targets]: I'd change the wording to something like "Could not find an installation of Visual Studio with the C++ development tools."

File: build/repo.targets [hunk 3]
```diff
@@ -4,26 +4,18 @@
...
-  <Target Name="BuildNativeAssets" DependsOnTargets="_LocateMSBuildExe" BeforeTargets="Compile">
+  <Target Name="BuildNativeAssets" DependsOnTargets="GetToolsets" BeforeTargets="Compile" >
     <ItemGroup>
       <BuildConfigurations Include="/p:Configuration=Release /p:platform=Win32" />
       <BuildConfigurations Include="/p:Configuration=Release /p:platform=x64" />
     </ItemGroup>
-    <Exec Command="&quot;$(MSBuildExePath)&quot; &quot;$(RepositoryRoot)src\AspNetCore\AspNetCore.vcxproj&quot; %(BuildConfigurations.Identity)" />
+
+    <Error
+      Text="Could not find an installation of Visual Studio with the C++ development tools. MSBuild path: $(VisualStudioMSBuildx86Path)"
```
Comment (natemcmaster) [inline build/repo.targets]: Doesn't really make sense to put $(VisualStudioMSBuildx86Path) in the error as it will always be empty due to the Condition.

**Review by natemcmaster — APPROVED** [review]
One suggestion. Changes look fine, although build fails on CI

**Pull request merged**